### PR TITLE
Create Playlist Operations

### PR DESCRIPTION
**Pull request by hugokishi**
**Context**

This PR includes a new functionality, now the user can create one or more playlists

### CODE REVIEW

File: src/interface/http/routes/api.ts [hunk 1]
```diff
@@ -27,6 +30,10 @@ class Routes {
   mountMusicRoutes () {
     this.router.post('/musics', MulterUpload.single('file'), (req, res) => new MusicController(req, res).store())
   }
+
+  mountPlaylistRoutes () {
+    this.router.post('/playlists', ThumbnailUpload.single('thumbnail'), (req, res) => new PlaylistController(req, res).store())
+  }
```
Comment (ramos-ph) [inline src/interface/http/routes/api.ts:36]: I don't think we'll need to create a Thumbnail upload.

Remember that the playlist thumbnail will be randomly selected based on the tracks it contains.

**Review by ramos-ph — CHANGES_REQUESTED** [review]

File: src/infrastructure/database/orm/entities/Music.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { Column, CreateDateColumn, Entity, Generated, PrimaryGeneratedColumn } f
 @Entity({ name: 'musics' })
 class Music {
   @PrimaryGeneratedColumn()
-  @Generated('uuid')
+  @Generated('increment')
```
[the remark this answers is not in the document]
Comment (hugokishi) [reply src/infrastructure/database/orm/entities/Music.ts:6]: For many to many relation in playlist

File: src/config/thumnail.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  storage: storage.local,
+  limits: {
+    fileSize: 2 * 4086 * 4086
+  },
+  fileFilter: (req, file, cb) => {
+    const allowedMimes = [
+      'image/jpeg',
+      'image/pjpeg',
+      'image/png',
+      'image/gif'
```
[the remark this answers is not in the document]
Comment (hugokishi) [reply src/config/thumnail.ts]: removed gif

File: src/application/use_cases/playlist/CreatePlaylist.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  public async execute () {
+    return this.playlistRepository.store({
+      name: this.name,
+      thumbnail: this.getThumnailUrl(this.thumbnail)
+    })
+  }
+
+  private getThumnailUrl (thumbnail: string): string {
```
[the remark this answers is not in the document]
Comment (hugokishi) [reply src/application/use_cases/playlist/CreatePlaylist.ts]: resolved

File: src/config/thumnail.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import multer from 'multer'
```
[the remark this answers is not in the document]
Comment (hugokishi) [reply src/config/thumnail.ts]: resolved, name changed

File: src/interface/http/routes/api.ts (hunk 1 above)
Comment (hugokishi) [reply src/interface/http/routes/api.ts:36]: I think the user should choose, if not, we set a default based on the songs

File: src/infrastructure/database/orm/repository/PlaylistRepository.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { getManager } from 'typeorm'
+import Playlist from '../../../../domain/Playlist/Playlist'
+import PlaylistEntityFactory from '../factories/PlaylistEntityFactory'
+import PlaylistRepository from '../../../../domain/Playlist/PlaylistRepository'
+
+class PlaylistRepositoryImpl implements PlaylistRepository {
+  async store (playlist: Playlist): Promise<Playlist> {
```
[the remark this answers is not in the document]
Comment (hugokishi) [reply src/infrastructure/database/orm/repository/PlaylistRepository.ts:7]: In this scenario the playlist could be created only after having a song, at first it is created empty

File: src/interface/http/routes/api.ts (hunk 1 above)
Comment (ramos-ph) [reply src/interface/http/routes/api.ts:36]: It should, actually.
But at the time of creation, we want the user to spend the less amount of time doing this.

Comment (ramos-ph) [reply src/interface/http/routes/api.ts:36]: What I was thinking is that at the time of creation, we ask to the user give the playlist a name and that's it.
Then the user could go to their playlists and update them at will. What do you think?

File: src/infrastructure/database/orm/entities/Music.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ramos-ph) [reply src/infrastructure/database/orm/entities/Music.ts:6]: Can't we use UUID for this? If not, we can have both IDs. Having a single numeric ID can give both corrupted data and performance issues.

[the remark this answers is not in the document]
Comment (ramos-ph) [reply src/infrastructure/database/orm/entities/Music.ts:6]: Just to clarify, with "both IDs" what I mean is that we can use the UUID for querying and the numeric for metadata to make the relations and joins.

For example `GET: /playlists/uuid/tracks`

File: src/infrastructure/database/orm/repository/PlaylistRepository.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (ramos-ph) [reply src/infrastructure/database/orm/repository/PlaylistRepository.ts:7]: In that case, it's fine. There's no need to add this rule on this PR. We can create another task for that

File: src/interface/http/routes/api.ts (hunk 1 above)
Comment (hugokishi) [reply src/interface/http/routes/api.ts:36]: I think it's a good one, I'll remove it here

File: src/infrastructure/database/orm/entities/Music.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (hugokishi) [reply src/infrastructure/database/orm/entities/Music.ts:6]: alright, resolved

**Review by ramos-ph — APPROVED** [review]
LGTM! Great work!

**Pull request merged**